query-sql/src/lib.rs

```diff
@@ -143,7 +148,7 @@ pub struct Join {
 pub enum TableOrSubquery {
     Table(SourceAlias),
     Union(Vec<SelectQuery>, TableAlias),
-    // TODO: Subquery.
+    Subquery(Rc<SelectQuery>),
```

Comment:
Without the `Rc` I get a recursion problem as `SelectQuery` contains a `FromClause` contains a `Join` which contains a `TableOrSubquery` which contains a `SelectQuery` which contains a `FromClause` and so on. `Rc` seemed to be the most appropriate way to fix this, but if you have another suggestion I am open to it.